lib/ExtensionPoint/pages/docs.php

```diff
@@ -45,19 +45,29 @@
         }
         $content .= '</nav>';
     } else {
+        
+        $editor = rex_editor::factory();
+        $editButton = null === $editor->getName()
+            ? null
+            : ' <a href="" class="btn btn-info btn-xs"><i class="fa fa-edit"></i> '.$editor->getName().'</a>';
+
         $toc = [];
         $extensionPoints = ExtensionPoint::getByPackage($requestIndex);
         foreach ($extensionPoints as $extensionPoint) {
             $toc[] = '<a href="#' . $extensionPoint->getName() . '">' . $extensionPoint->getName() . '</a>';
 
+            $button = null === $editButton
+                ? ''
+                : substr_replace( $editButton, $editor->getUrl($extensionPoint->getFilepath(),$extensionPoint->getLn()), 10, 0 );
```

Comment:
ich denke statt 2x markup zu bauen könnte man direkt hier das ganze in einem zug machen (ohne substr_replace() etc

Reply:
Und wo fängst Du ab, wenn der Editor gar nicht aktiviert ist? Also so, dass es noch halbwegs übersichtlich ist?

Reply:
ungeteste so:

```php
$button = '';

$editor = rex_editor::factory();
if ($url = $editor->getUrl($extensionPoint->getFilepath(),$extensionPoint->getLn())) {
  $button = ' <a href="'. $url .'" class="btn btn-info btn-xs"><i class="fa fa-edit"></i> '.$editor->getName().'</a>';
}
```

Reply:
Ich hab das mal aufgenommen und den PR angepasst. Allerdings hab ich weiter `$editor = rex_editor::factory();` vor der Schleife gelassen.